src/AWS.Deploy.Orchestration/Utilities/CloudApplicationNameGenerator.cs

```diff
@@ -105,5 +121,20 @@ public string InvalidNameMessage(string name)
             return $"Invalid cloud application name {name}. The application name can contain only alphanumeric characters (case-sensitive) and hyphens. " +
                 "It must start with an alphabetic character and can't be longer than 128 characters";
         }
+
+        public bool MatchesExistingDeployment(string name, DeploymentTypes deploymentType, IList<CloudApplication> existingApplications)
+        {
+            if (!existingApplications.Any())
+                return false;
+
+            var filteredApplications = existingApplications.Where(x => x.DeploymentType == deploymentType);
+            return filteredApplications.Any(x => x.Name == name);
+        }
+
+        public string ApplicationNameAlreadyExistsMessage(string name)
+        {
+            return $"Invalid cloud application name {name}. There already exists a cloud application name you provided. " +
+                "Please choose another cloud application name.";
```

Comment:
Suggestion: 
Invalid cloud application name: {name}. A cloud application already exists with this name.

Reply:
Fixed

Reply:
I didn't see a change to this string. Can you confirm that an updated was pushed to the PR?

Reply:
Added the correct suggestion. Thank you for pointing it out.